common/include/algebra/type_traits.hpp

```diff
@@ -129,7 +131,61 @@ template <class M>
 using block_getter_t = typename block_getter<M>::type;
 /// @}
 
-}  // namespace algebra::traits
+/// The algebra types
+/// @{
+template <typename T>
+struct get_algebra {};
+
+template <typename T>
+requires(!std::is_same_v<typename T::point3D, void>) struct get_algebra<T> {
+  template <typename U>
+  using simd = typename T::template simd<U>;
+  using size_type = typename T::size_type;
+  using boolean = typename T::boolean;
+  using value = typename T::value_type;
+  using scalar = typename T::scalar;
+  using point2D = typename T::point2D;
+  using point3D = typename T::point3D;
+  using vector3D = typename T::vector3D;
+  using transform3D = typename T::transform3D;
+  template <std::size_t ROWS, std::size_t COLS>
+  using matrix = typename T::template matrix<ROWS, COLS>;
+};
+
+}  // namespace traits
+
+template <typename A>
+using get_value_t = typename traits::get_algebra<A>::value;
```

Comment:
Should we use the `get_` prefix for all these things? I'm not sure myself...

Reply:
I don't know either tbh. They are "typedefed away" in the detray namespace, so nobody will really see them I think